pkg/policy/api/rule_validation.go

```diff
@@ -78,20 +80,27 @@ func (i *IngressRule) sanitize() error {
 		}
 	}
 
-	if l := len(i.FromCIDR); l > MaxCIDREntries {
-		return fmt.Errorf("too many ingress CIDR entries %d/%d", l, MaxCIDREntries)
-	}
-
+	prefixLengths := map[int]exists{}
 	for n := range i.FromCIDR {
-		if err := i.FromCIDR[n].sanitize(); err != nil {
+		prefixLength, err := i.FromCIDR[n].sanitize()
+		if err != nil {
 			return err
 		}
+		prefixLengths[prefixLength] = exists{}
 	}
 
 	for n := range i.FromCIDRSet {
-		if err := i.FromCIDRSet[n].sanitize(); err != nil {
+		prefixLength, err := i.FromCIDRSet[n].sanitize()
+		if err != nil {
 			return err
 		}
+		prefixLengths[prefixLength] = exists{}
+	}
+
+	// FIXME GH-1781 count coalesced CIDRs and restrict the number of
+	// prefix lengths based on the CIDRSet exclusions.
+	if l := len(prefixLengths); l > MaxCIDRPrefixLengths {
```

Comment:
I think this is not particularly useful, because the number that matters is the number of CIDRs across all the rules relevant for each specific endpoint.
It's OK to check here anyway.

Reply:
Do you think we should just drop this validation and rely on the generation errors propagating to endpoint log?

Reply:
We can keep it here, there's also checking when L3Policy is `Validate()`d just before attempting to push it into BPF.